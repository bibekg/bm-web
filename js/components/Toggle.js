// @flow

import * as React from 'react'
import styled from 'styled-components'
import { colors } from 'styles'

const ToggleFrame = styled.div`
  width: 44px;
  height: 25px;
  border: 3px solid ${colors.blue};
  background: ${colors.blue};
  border-radius: 6px;
  column-count: 2;
  column-gap: 0px;
  cursor: pointer;
  opacity: ${props => (props.toggled ? 1 : 0.5)};
`

const ToggleInner = styled.div`
  width: 100%;
  height: 100%;
  border-radius: 4px;
  background-color: white;
  opacity: ${props => (props.toggled ? 1 : 0)};
  transition: opacity 0.3s ease;
`

type PropsType = {
  toggled: boolean,
  onClick: (SyntheticInputEvent<*>) => void
}

export default function Toggle(props: PropsType): React.Element<*> {
  return (
    <ToggleFrame toggled={props.toggled} onClick={props.onClick}>
      <ToggleInner toggled={!props.toggled} />
      <ToggleInner toggled={props.toggled} />
    </ToggleFrame>
  )
}
