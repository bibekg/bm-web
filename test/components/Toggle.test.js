import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Toggle from 'components/Toggle'

describe('<Toggle />', () => {
  test('should render correctly with no options', () => {
    const wrapper = shallow(<Toggle />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should render correctly when toggled off', () => {
    const wrapper = shallow(<Toggle toggled={false} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should handle onClick correctly', () => {
    const clickFcn = jest.fn()
    const wrapper = shallow(<Toggle toggled={false} onClick={clickFcn} />)

    wrapper.simulate('click')
    expect(clickFcn).toHaveBeenCalled()
  })
})
