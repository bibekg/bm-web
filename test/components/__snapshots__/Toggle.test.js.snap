// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should render correctly when toggled off 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.div
    enabled={true}
    left={true}
  />
  <styled.div />
</styled.div>
`;

exports[`<Toggle /> should render correctly with no options 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.div
    enabled={true}
    left={true}
  />
  <styled.div />
</styled.div>
`;
